kommunicateui/src/main/java/com/applozic/mobicomkit/uiwidgets/conversation/fragment/FeedbackInputFragment.java

```diff
@@ -0,0 +1,245 @@
+
+        feedbackRatingGroup = new FeedbackRatingGroup(3, buttonDrawablesSelected, buttonDrawablesUnselected);
+
+        feedbackRatingGroup.createViewForRatingValue(view, 1, R.id.idButtonPoor, R.id.idTextPoor);
+        feedbackRatingGroup.createViewForRatingValue(view, 2, R.id.idButtonAverage, R.id.idTextAverage);
+        feedbackRatingGroup.createViewForRatingValue(view, 3, R.id.idButtonGood, R.id.idTextGood);
+
+        buttonSubmitFeedback.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                if(ratingLevel == 0) {
+                    setRatingLevel(RATING_AVERAGE);
+                }
+                String feedbackComment = editTextFeedbackComment.getText().toString().trim();
+                feedbackFragmentListener.onFeedbackSubmitButtonPressed(getRatingLevel(), feedbackComment);
+                getFragmentManager().popBackStack();
+            }
+        });
+
+        buttonCloseFragment.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                getFragmentManager().popBackStack();
+            }
+        });
+
+        return view;
+    }
+
+    /**
+     * toggle button as selected or deselected
+     * @param select true or false
+     * @param feedbackRating the views to toggle
+     */
+    void toggleRatingButtonSelected(boolean select, FeedbackRatingGroup.FeedbackRating feedbackRating) {
+        //show the feedback comment input edit text, if not already visible
+        if(editTextFeedbackComment.getVisibility()==View.GONE) {
+            editTextFeedbackComment.setVisibility(View.VISIBLE);
+        }
+
+        //set drawable
+        feedbackRatingGroup.setDrawableForValue(select, feedbackRating.ratingValue);
+
+        //set visibility and size
+        if(select) {
+            feedbackRating.ratingButton.setScaleX(1f);
+            feedbackRating.ratingButton.setScaleY(1f);
+            feedbackRating.feedbackTextView.setVisibility(View.VISIBLE);
+        } else {
+
+            feedbackRating.ratingButton.setScaleX(0.8f);
+            feedbackRating.ratingButton.setScaleY(0.8f);
+            feedbackRating.feedbackTextView.setVisibility(View.GONE);
+        }
+    }
+
+    //when one of the rating buttons is clicked
+    @Override
+    public void onClick(View view) {
+        Integer buttonTag = (Integer) view.getTag();
+
+        for(FeedbackRatingGroup.FeedbackRating feedbackRating : feedbackRatingGroup.feedbackRating) {
+            Integer iterationButtonTag = (Integer) feedbackRating.ratingButton.getTag();
+            if(iterationButtonTag.intValue() == buttonTag.intValue()) {
+                setRatingLevel(buttonTag);
+                toggleRatingButtonSelected(true, feedbackRating);
+            }
+            else
+                toggleRatingButtonSelected(false, feedbackRating);
```

Comment:
@shubhamtewari we already have if - else here and we are passing true, false and then inside toggleRatingButtonSelected again we have if - else.
it will be better if instead of toggle method, we create 2 methods one for selected state and another for non selected state and directly call if from this if - else.

Reply:
moved if expression inside function signature to remove if-else

Reply:
done